File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/algorithms/rotor_router_with_leader_dispersion/models/RotorRouterWithLeaderDispersionRobot.java
package hu.elteik.knowledgelab.javaengine.algorithms.rotor_router_with_leader_dispersion.models;

import hu.elteik.knowledgelab.javaengine.core.models.Robot;
import hu.elteik.knowledgelab.javaengine.core.utils.Color;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper=false)
@AllArgsConstructor
@NoArgsConstructor
public class RotorRouterWithLeaderDispersionRobot extends Robot {
    
    private Color color;
    private Long onID;
    private Long destinationID;
    private Long lastUsedEdgeID;

}

File: dispersion-app/src/app/models/algorithms/rotor-router-with-leader-dispersion/RotorRouterWithLeaderDispersionRobot.ts
import { Robot } from '../../core/Robot';
import { RobotState } from '../../utils/RobotState';

export class RotorRouterWithLeaderDispersionRobot extends Robot {
  onID: number;
  destinationID: number;
  lastUsedEdgeID: number;

  constructor(
    id: number,
    state: RobotState,
    onID: number,
    destinationID: number,
    lastUsedEdgeID: number
  ) {
    super(id, state);
    this.onID = onID;
    this.destinationID = destinationID;
    this.lastUsedEdgeID = lastUsedEdgeID;
  }
}
